### PR TITLE
ColorManagement: Add .getLuminanceCoefficients

### PR DESCRIPTION
**Pull request by donmccurdy**
Follow-up to #28861. Adds a method to ColorManagement, to produce luminance coefficients for a given color space. Currently supports sRGB, Linear sRGB, Display P3, and Linear Display P3. For sRGB and Linear sRGB I used the (more common?) 4-decimal precision, `0.2126, 0.7152, 0.0722`, simply or consistency with OCIO, Blender, and ASC specifications. I expect the difference will be ~zero, but if we see E2E failures as a result, I can switch back. ~~I believe it's fair to call these either "luma coefficients" or "luminance coefficients", I don't have a preference other than internal consistency.~~

Certain post-processing effects were using luminance coefficients from Rec. 601, which we do not otherwise support. I've corrected those coefficients.

~~One notable exception on ColorManagement's use, `common.glsl.js` contains hard-coded luminance coefficients. Ideally they'd be based on the working color space, but I'm unsure if it's a good idea to put a new global uniform into this shader chunk. Suggestions welcome. In terms of working toward support for wide-gamut PBR rendering, it might be more practical to focus on WebGPURenderer for now.~~ Fixed. ✅

Finally, note that while the luminance coefficients are the same for Linear sRGB vs. sRGB, and for Linear Display P3 vs. Display P3, the dot product gives [different results with different meanings](https://cs.stackexchange.com/a/92589) given linear vs. non-linear components.

/cc @Mugen87 @WestLangley

### CODE REVIEW

File: examples/jsm/interactive/HTMLMesh.js [hunk 1]
```diff
@@ -371,7 +374,8 @@ function html2canvas( element ) {
 
 				color.set( accentColor );
 
-				const luminance = Math.sqrt( 0.299 * ( color.r ** 2 ) + 0.587 * ( color.g ** 2 ) + 0.114 * ( color.b ** 2 ) );
+				ColorManagement.getLuminanceCoefficients( lumaCoeffs );
+				const luminance = Math.sqrt( lumaCoeffs.x * ( color.r ** 2 ) + lumaCoeffs.y * ( color.g ** 2 ) + lumaCoeffs.z * ( color.b ** 2 ) );
```
Comment (WestLangley) [inline examples/jsm/interactive/HTMLMesh.js:378]: Do you have any idea what this is computing? I'm not sure you should worry about this code chunk...

**Comment by WestLangley** [comment]
>I believe it's fair to call these either "luma coefficients" or "luminance coefficients", I don't have a preference other than internal consistency.

Can we please use `luninanceCoefficients` and stay away from `luma`? `luninanceCoeffs` is OK, too, if you want.

**Comment by Mugen87** [comment]
>  Ideally they'd be based on the working color space, but I'm unsure if it's a good idea to put a new global uniform into this shader chunk. Suggestions welcome.

I would say yes so ideally components like `LuminosityHighPassShader` could directly use the `luminance()` function from `commons.glsl`. That should make the code more readable.

When thinking about it we can have a solution without a uniform at all if you update `WebGLProgram` instead. You can embed the coefficients directly in the fragment shader similar to the tone mapping and color space logic.

File: examples/jsm/interactive/HTMLMesh.js (hunk 1 above)
Comment (donmccurdy) [reply examples/jsm/interactive/HTMLMesh.js:378]: Asking for clarification in https://github.com/mrdoob/three.js/pull/24030/files#r1679662069, but I'd be OK with skipping this file too.

**Comment by donmccurdy** [comment]
> Can we please use luninanceCoefficients and stay away from luma? luninanceCoeffs is OK, too, if you want.

Ok, works for me. 👍🏻 

> You can embed the coefficients directly in the fragment shader similar to the tone mapping and color space logic.

Will look into this – thanks! 

Moving this PR to 'draft' as I address feedback.

File: examples/jsm/interactive/HTMLMesh.js (hunk 1 above)
Comment (donmccurdy) [reply examples/jsm/interactive/HTMLMesh.js]: Let's skip HTMLMesh.js for now, as this chunk of code is otherwise unrelated to the rendering pipeline.

**Comment by donmccurdy** [comment]
I believe all feedback has been addressed. ✅

File: src/nodes/display/ColorAdjustmentNode.js [hunk 2]
```diff
@@ -86,8 +88,8 @@ export const saturation = nodeProxy( ColorAdjustmentNode, ColorAdjustmentNode.SA
 export const vibrance = nodeProxy( ColorAdjustmentNode, ColorAdjustmentNode.VIBRANCE );
 export const hue = nodeProxy( ColorAdjustmentNode, ColorAdjustmentNode.HUE );
 
-export const lumaCoeffs = vec3( 0.2126729, 0.7151522, 0.0721750 ); // // assumes rgb is in linear color space with sRGB primaries and D65 white point
-export const luminance = ( color, luma = lumaCoeffs ) => dot( color, luma );
+const luminanceCoefficients = /*#__PURE__*/ new Vector3();
+export const luminance = ( color, luma = vec3( ... ColorManagement.getLuminanceCoefficients( luminanceCoefficients ) ) ) => dot( color, luma );
```
Comment (WestLangley) [inline src/nodes/display/ColorAdjustmentNode.js:92]: Can we avoid the word `luma`. How about `coeffs`, or preferably, `luminanceCoefficients`?

Comment (donmccurdy) [reply src/nodes/display/ColorAdjustmentNode.js]: Done!

**Review by Mugen87 — APPROVED** [review]

**Pull request merged**